### ISSUE TITLE
Labels added in v0.12.1 prevents applying changes to deployment

### ISSUE THREAD

**Issue by dewe** [bitnami-labs/sealed-secrets#395]
Deployment label selectors are immutable, and the new label selector prevents us from deploying changes via our normal gitops pipeline.

Error from kubectl (in Flux):
```
 The Deployment "sealed-secrets-controller" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app.kubernetes.io/managed-by":"jsonnet", "app.kubernetes.io/name":"kubeseal", "app.kubernetes.io/part-of":"kubeseal", "app.kubernetes.io/version":"v0.12.1", "name":"sealed-secrets-controller"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable
```

This affects anyone using `kubectl apply -f`, and I presume this would happen with every version upgrade because of app.kubernetes.io/version changing.

**Comment by mkmik** [comment]
Uh, this seems serious. High prio

**Comment by dewe** [comment]
Fixed in v0.12.2. Thanks.

**Comment by mkmik** [comment]
thanks for updating the issue!
